subsys/fs/littlefs_fs.c

```diff
@@ -721,6 +722,11 @@ static int littlefs_unmount(struct fs_mount_t *mountp)
 	return 0;
 }
 
+static bool littlefs_compatible(const char *type_sz)
+{
+	return (strcmp(type_sz, "LittleFS") == 0);
```

Comment:
The canonical spelling is `littlefs`, even though there are some remaining uses of `LittleFS` from when I was ignorant.

I'd use `strcasecmp`, though, and document that comparisons are case-insensitive in the API.  No point in annoying users who happen to guess wrong on the capitalization pattern.